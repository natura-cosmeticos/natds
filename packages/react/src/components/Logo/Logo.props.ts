import { Color, Size } from '@naturacosmeticos/natds-themes'

type LogoColors = 'neutral' | keyof Pick<Color, 'primary' | 'secondary' | 'highlight' | 'surface'>
type LogoSizes = keyof Omit<Size, 'none' | 'micro' | 'tiny' | 'small' | 'standard' | 'semi' | 'semiX'>

export interface LogoProps {
  /**
   * The Logo description
   */
  arialLabel?: string

  /**
  * The Logo color
  * @default neutral
  */
  color?: LogoColors;

  /**
   * The Logo model.
   * @default primary
   */
  model?: LogoModels;

  /**
   * The Logo size
   * @default veryhuge
   */
  size?: LogoSizes
}
