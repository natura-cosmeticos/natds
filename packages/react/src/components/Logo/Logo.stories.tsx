import React from 'react'
import { Story, Meta } from '@storybook/react'
import { Logo, LogoProps } from '.'
import StoryContainer from '../../helpers/StoryContainer'

const componentStatus = `
---

Attribute status:

- **Size**
    - ✅ \`medium\`
    - ✅ \`mediumx\`
    - ✅ \`large\`
    - ✅ \`largex\`
    - ✅ \`largexx\`
    - ✅ \`largexxx\`
    - ✅ \`huge\`
    - ✅ \`hugex\`
    - ✅ \`hugexx\`
    - ✅ \`hugexxx\`
    - ✅ \`veryhuge\`
- **Color**
    - ✅ \`neutral\`
    - ❌ \`primary\`
    - ❌ \`secondary\`
    - ❌ \`highlight\`
    - ❌ \`surface\`
- **Model**
    - ✅ \`primary\`
    - ❌ \`secondary\`

---
`

export default {
  title: 'Components/Logo',
  component: Logo,
  parameters: {
    componentSubtitle: 'Logo adapted as brand standards for each company',
    docs: { description: { component: componentStatus } },
    actions: { argTypesRegex: '^on.*' }
  }
} as Meta

export const Playground: Story<LogoProps> = (args) => <Logo {...args} />
Playground.args = { size: 'large' }

export const Size: Story<LogoProps> = (args) => (
  <StoryContainer>
    <Logo {...args} size="medium" />
    <Logo {...args} size="mediumX" />
    <Logo {...args} size="large" />
    <Logo {...args} size="largeX" />
    <Logo {...args} size="largeXX" />
    <Logo {...args} size="largeXXX" />
    <Logo {...args} size="huge" />
    <Logo {...args} size="hugeX" />
    <Logo {...args} size="hugeXX" />
    <Logo {...args} size="hugeXXX" />
    <Logo {...args} size="veryHuge" />
  </StoryContainer>
)

export const Color: Story<LogoProps> = (args) => (
  <StoryContainer>
    <Logo {...args} />
  </StoryContainer>
)
Color.args = { ...Playground.args }

export const Model: Story<LogoProps> = (args) => (
  <StoryContainer>
    <Logo {...args} model="primary" />
    <Logo {...args} model="secondary" />
  </StoryContainer>
)
Model.args = { ...Playground.args }
