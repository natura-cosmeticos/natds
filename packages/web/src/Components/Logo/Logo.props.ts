import { ISizes } from '@naturacosmeticos/natds-styles'
import { Color } from '@naturacosmeticos/natds-themes'

type LogoColors = 'neutral' | keyof Pick<Color, 'primary' | 'secondary' | 'highlight' | 'surface'>
type LogoSizes = keyof Omit<ISizes, 'none' | 'micro' | 'tiny' | 'small' | 'standard' | 'semi' | 'semiX'>
type LogoModels = 'a' | 'b'

export interface ILogoProps {
  /**
   * The Logo description
   */
  arialLabel?: string

  /**
  * The Logo color
  * @default neutral
  */
  color?: LogoColors;

  /**
   * The Logo model.
   * @default a
   */
  model?: LogoModels;

  /**
   * The Logo size
   * @default veryhuge
   */
  size?: LogoSizes
}
