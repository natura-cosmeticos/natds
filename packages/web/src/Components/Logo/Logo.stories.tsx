import React from 'react'
import { Story, Meta } from '@storybook/react'
import { Logo, LogoProps } from '.'

export default {
  title: 'Components/Logo',
  component: Logo,
  parameters: {
    componentSubtitle: 'Logo adapted as brand standards for each company',
    actions: { argTypesRegex: '^on.*' }
  }
} as Meta

export const Playground: Story<LogoProps> = (args) => <Logo {...args} />

export const Size: Story<LogoProps> = (args) => (
  <>
    <Logo {...args} size="medium" />
    <Logo {...args} size="mediumX" />
    <Logo {...args} size="large" />
    <Logo {...args} size="largeX" />
    <Logo {...args} size="largeXX" />
    <Logo {...args} size="largeXXX" />
    <Logo {...args} size="huge" />
    <Logo {...args} size="hugeX" />
    <Logo {...args} size="hugeXX" />
    <Logo {...args} size="hugeXXX" />
    <Logo {...args} size="veryHuge" />
  </>
)

export const Color: Story<LogoProps> = (args) => (
  <>
    <Logo {...args} />
    <Logo {...args} color="primary" />
    <Logo {...args} color="secondary" />
    <Logo {...args} color="highlight" />
    <Logo {...args} color="surface" />
  </>
)

export const Model: Story<LogoProps> = (args) => (
  <>
    <Logo {...args} />
  </>
)
Model.args = { ...Playground.args, model: 'a' }
