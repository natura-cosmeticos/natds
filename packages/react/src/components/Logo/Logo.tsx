import React from 'react'
import { Theme } from '@naturacosmeticos/natds-themes'
import { useTheme } from 'react-jss'
import { LogoProps } from './Logo.props'
import styles from './Logo.styles'

const Logo = ({
  color = 'neutral',
  size = 'veryHuge',
  arialLabel = 'logo'
}: LogoProps): JSX.Element => {
  const theme: Theme = useTheme()
  const classes = styles({ size, color })

  const checkColor = color === 'neutral' ? 'neutral' : 'custom'
  const checkModel = model === 'primary' ? 'a' : 'b'

  const logo = theme.asset.brand[checkColor][checkModel]

  return (
    <div
      className={classes.root}
      dangerouslySetInnerHTML={{ __html: logo }}
      role="img"
      aria-label={arialLabel}
    />
  )
}

export default Logo
