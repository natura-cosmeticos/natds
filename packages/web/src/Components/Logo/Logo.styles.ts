import createStyles from '../../styles/createStyles'
import { makeStyles } from '../../styles/makeStyles'
import { LogoProps } from './Logo.props'
import { IThemeWeb } from '../../Themes'

const styles = ({ sizes, color: pallete }: IThemeWeb) => createStyles({
  root: {
    '& svg': {
      height: '100%',
      width: ({ size }: LogoProps) => size && sizes[size],
      '& path': {
        fill: ({ color }: LogoProps) => color !== 'neutral' && color && pallete[color]
      }
    }
  }
})

export const useStyles = makeStyles(styles, { name: 'NatDSLogo' })
export default useStyles
