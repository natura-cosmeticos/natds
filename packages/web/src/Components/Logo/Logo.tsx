/* eslint-disable react/no-danger-with-children */
import React from 'react'
import { useTheme } from '@material-ui/core/styles'
import { IThemeWeb } from '../../Themes'
import { LogoProps } from './Logo.props'
import { useStyles } from './Logo.styles'

const Logo = ({
  color = 'neutral',
  size = 'veryHuge',
  model = 'a',
  arialLabel = 'logo'
}: LogoProps) => {
  const theme: IThemeWeb = useTheme()
  const { root } = useStyles({ size, color })

  const checkColor = color === 'neutral' ? 'neutral' : 'custom'

  console.log({ theme })

  const logo = theme.asset.brand[checkColor][model].file

  return (
    <div
      style={{ width: theme.sizes[size] }}
      className={root}
      dangerouslySetInnerHTML={{ __html: logo }}
      role="img"
      aria-label={arialLabel}
    />
  )
}

Logo.displayName = 'Logo'

export default Logo
