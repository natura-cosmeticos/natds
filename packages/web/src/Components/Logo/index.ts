import Logo from './Logo'

export { Logo }
export { ILogoProps } from './Logo.props'
export default Logo
