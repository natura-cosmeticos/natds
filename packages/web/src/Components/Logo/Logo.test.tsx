import React from 'react'
import * as TestRenderer from 'react-test-renderer'
import Logo from './Logo'

describe('Logo component', () => {
  it('should render correctly with default props', () => {
    const component = TestRenderer.create(<Logo />)

    expect(component).toMatchSnapshot()
  })
  it('should render correctly with medium size', () => {
    const component = TestRenderer.create(<Logo size="medium" />)

    expect(component).toMatchSnapshot()
  })
  it('should render correctly with custom color', () => {
    const component = TestRenderer.create(<Logo color="primary" />)

    expect(component).toMatchSnapshot()
  })

  it('should render correctly with secondary model', () => {
    const component = TestRenderer.create(<Logo model="a" />)

    expect(component).toMatchSnapshot()
  })
})
