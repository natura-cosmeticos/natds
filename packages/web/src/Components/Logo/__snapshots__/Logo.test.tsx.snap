// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component should render correctly with custom color 1`] = `
<div
  aria-label="logo"
  className="NatDSLogo-root-1 NatDSLogo-root-4"
  dangerouslySetInnerHTML={
    Object {
      "__html": undefined,
    }
  }
  role="img"
/>
`;

exports[`Logo component should render correctly with default props 1`] = `
<div
  aria-label="logo"
  className="NatDSLogo-root-1 NatDSLogo-root-2"
  dangerouslySetInnerHTML={
    Object {
      "__html": undefined,
    }
  }
  role="img"
/>
`;

exports[`Logo component should render correctly with medium size 1`] = `
<div
  aria-label="logo"
  className="NatDSLogo-root-1 NatDSLogo-root-3"
  dangerouslySetInnerHTML={
    Object {
      "__html": undefined,
    }
  }
  role="img"
/>
`;

exports[`Logo component should render correctly with secondary model 1`] = `
<div
  aria-label="logo"
  className="NatDSLogo-root-1 NatDSLogo-root-5"
  dangerouslySetInnerHTML={
    Object {
      "__html": undefined,
    }
  }
  role="img"
/>
`;
